source/gfoidl.DataCompression/Compression/SwingingDoorCompression/SwingingDoorCompressionIterator.cs

```diff
@@ -34,11 +34,34 @@ protected void CloseTheDoor(in DataPoint incoming, in DataPoint lastArchived)
         {
             Debug.Assert(_swingingDoorCompression is not null);
 
-            double upperSlope = lastArchived.Gradient(incoming,  _swingingDoorCompression.CompressionDeviation);
-            double lowerSlope = lastArchived.Gradient(incoming, -_swingingDoorCompression.CompressionDeviation);
+            double delta_x = incoming.X - lastArchived.X;
 
-            if (upperSlope < _slope.Max) _slope.Max = upperSlope;
-            if (lowerSlope > _slope.Min) _slope.Min = lowerSlope;
+            if (delta_x != 0)
```

Comment:
```suggestion
            if (delta_x > 0)
```
Is this better? Can't be less than 0, so maybe this elides another jmp in the assembly.

Reply:
Yes, it is.

```diff
       vzeroupper
       lea       rdx,[rcx+30]
       lea       r8,[rcx+20]
       vmovsd    xmm0,qword ptr [rdx]
       vsubsd    xmm0,xmm0,qword ptr [r8]
       vxorps    xmm1,xmm1,xmm1
       vucomisd  xmm0,xmm1
-      jp        short M00_L00
-      je        short M00_L03
+      jbe       short M00_L03
M00_L00:
       vmovsd    xmm1,qword ptr [rdx+8]
       vsubsd    xmm1,xmm1,qword ptr [r8+8]
       mov       rdx,[rcx+8]
       vmovsd    xmm2,qword ptr [rdx+38]
       vaddsd    xmm3,xmm1,xmm2
       vdivsd    xmm3,xmm3,xmm0
       vsubsd    xmm1,xmm1,xmm2
       vdivsd    xmm0,xmm1,xmm0
       vmovsd    xmm1,qword ptr [rcx+10]
       vucomisd  xmm1,xmm3
       jbe       short M00_L01
       vmovsd    qword ptr [rcx+10],xmm3
M00_L01:
       vucomisd  xmm0,qword ptr [rcx+18]
       jbe       short M00_L02
       vmovsd    qword ptr [rcx+18],xmm0
M00_L02:
       ret
M00_L03:
       jmp       near ptr gfoidl.DataCompression.Benchmarks.SwingingDoorCompressionIteratorCloseTheDoorBenchmarks.<Inlined2>g__GradientEquality|12_0(gfoidl.DataCompression.DataPoint ByRef, gfoidl.DataCompression.DataPoint ByRef)
-; Total bytes of code 102
+; Total bytes of code 100
```

Reply:
Maybe a similar optimization should be done in https://github.com/gfoidl/DataCompression/blob/6dc1da9f3fe1a816d185b382775cce628576e60d/source/gfoidl.DataCompression/DataPoint.cs#L90-L110 too.

* flip the branch condition
* make the uncommon path a tail call jmp